lib/grape/dsl/routing.rb

```diff
@@ -77,9 +77,12 @@ def do_not_route_options!
           namespace_inheritable(:do_not_route_options, true)
         end
 
-        def mount(mounts)
+        def mount(mounts, opts = {})
           mounts = { mounts => '/' } unless mounts.respond_to?(:each_pair)
           mounts.each_pair do |app, path|
+            if app.respond_to?(:mount_instance)
+              return mount(app.mount_instance(configuration: opts[:with] || {}) => path)
```

Comment:
The return issue is still not addressed, I still need a failing spec to see whether `next` is appropiate, or whether this is better handled by a refactor